### PR TITLE
Add integration tests and fix nxos providers

### PR DESCRIPTION
**Pull request by saichint**
##### SUMMARY
Fixes 
https://github.com/ansible/ansible/issues/26911
https://github.com/ansible/ansible/issues/27182
https://github.com/ansible/ansible/issues/27168
https://github.com/ansible/ansible/issues/27146
https://github.com/ansible/ansible/issues/27140
https://github.com/ansible/ansible/issues/27134

- Adds integration tests
- Fixes nxapi providers issue

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
nxos_evpn_vni
nxos_vtp_domain
nxos_vtp_password
nxos_vtp_version
nxos_vrrp
nxos_vrf
nxos_vrf_interface
nxos_rollback

##### ANSIBLE VERSION
```
ansible 2.4.0 (devel b49be56c5b) last updated 2017/07/13 12:38:51 (GMT -400)
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/p2/Projects/nxos_ansible/fix_ansible/lib/ansible
  executable location = /root/p2/Projects/nxos_ansible/fix_ansible/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```

##### ADDITIONAL INFORMATION
- Unit tests results:
```
================================================================== test session starts ==================================================================
platform linux -- Python 3.5.2, pytest-3.1.2, py-1.4.34, pluggy-0.4.0
rootdir: /root/p2/Projects/nxos_ansible/fix_ansible, inifile:
collected 3 items

test_nxos_evpn_vni.py ...

=============================================================== 3 passed in 0.09 seconds ================================================================
```

### CODE REVIEW

**Comment by saichint** [comment]
I did rebasing using https://docs.ansible.com/ansible/dev_guide/developing_rebasing.html
```
git pull --rebase upstream devel
From https://github.com/ansible/ansible
 * branch            devel      -> FETCH_HEAD
Current branch devel is up to date.
```
After that I did git push --force as well.
But it still shows 'needs_rebase' label. Not sure what else I need to do.
Any help from anyone?

**Comment by saichint** [comment]
never mind, it got removed now.

**Review by trishnaguha — APPROVED** [review]

**Comment by trishnaguha** [comment]
@saichint mind adding the test task in here https://github.com/ansible/ansible/blob/devel/test/integration/nxos.yaml ?
```
    - block:
      - include_role:
          name: nxos_evpn_vni
        when: "limit_to in ['*', 'nxos_evpn_vni']"
      rescue:
        - set_fact: test_failed=true
```

**Comment by saichint** [comment]
Added test task to nxos.yaml

**Comment by saichint** [comment]
@trishnaguha made changes as you suggested. if there are no more comments, could you merge it?
Thanks

File: lib/ansible/modules/network/nxos/nxos_vrf.py [hunk 1]
```diff
@@ -106,11 +106,10 @@
 def execute_show_command(command, module):
     transport = module.params['transport']
     if transport == 'cli':
```
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_vrf.py]: I would suggest removing the transport check (just so it is clear that the intention is to always ask for JSON), and, as we have this function gatekeeping all the `run_commands()` calls, only annotating the commands with `'output': 'text'` or `'output': 'json'` here. What is here works, but is inconsistent both in process (`'output': 'text'` vs `| json`) and location. This also avoids the type check.

**Review by Qalthos — CHANGES_REQUESTED** [review]

**Comment by saichint** [comment]
@Qalthos Made the changes. Let me know of any more comments

**Review by Qalthos — APPROVED** [review]

**Pull request merged**